### DISCUSSION TITLE
How to keep running apollo subscription while capacitor app is on background?

### DISCUSSION THREAD

**Discussion by enes-get-local** [ionic-team/capacitor#5376] [Q&A]
I been developing a react.js chat application which works via websocket connection with apollo-client.It uses capacitor.js. In chat component, a subscription query starts works and listen new for new messages. The problem is that messages are not listening while the app is  backgrounded. New messages starts loaded as soon as open app.I want to make it like whatsapp which load new messages while even app is on background. Is there a way to achieve this with capacitor ?

Apollo client configuration
```
const wsLink = process.browser
  ? new SubscriptionClient(WEBSOCKET_URL, {
      reconnect: true,
      connectionParams: () => ({
        headers: {
          authorization: `Bearer ${readLocalStorage('id_token')}`,
          'x-hasura-admin-secret':
            'xxxx',
        },
      }),
    })
  : null;
```

Component which subscription works
```
const onSubscriptionData = useCallback(
    ({ subscriptionData: { data } }: OnSubscriptionDataOptions<any>) => {
      if (selectedChatId.current) {
        setMessages([...(data.conversation || {})]);
        markAsRead();
      }
    },
    [selectedChatId.current]
  );

  const { loading: messagesLoading } = useSubscription(MESSAGE_DETAIL_SUBSCRIPTION, {
    onSubscriptionData,
    variables: {
      chat_thread_id: selectedChatId,
      user_id: user_id,
    },
  });
```

**Comment by bbjoern** [comment]
@enes-get-local  did you ever get it working ? i'm stuck with the same question

**Reply by mlynch** [reply]
The Web View does not run in the background, so this is not possible without app-store unapproved workarounds. You'll want to use Apollo's native iOS or Android SDKs and use the background functionality available there to handle this and then sync it back up with the web layer when the app comes back into the foreground

**Reply by bbjoern** [reply]
@mlynch thanks for the fast reply! What about the websocket connection (aka subscription). As far as I understand it will disconnect when the app goes to background and I will need to reconnect the connection when it comes back into foreground. Am I right?
